plugins/win-dshow/CMakeLists.txt

```diff
@@ -78,49 +78,51 @@ target_compile_definitions(
                     _CRT_NONSTDC_NO_WARNINGS)
 
 set(VIRTUALCAM_AVAILABLE OFF)
-if(VIRTUALCAM_GUID STREQUAL "")
-  obs_status(
-    WARNING
-    "Windows Virtual Camera - GUID not set - specify as 'VIRTUALCAM_GUID' to enable."
-  )
-else()
-  set(INVALID_GUID ON)
-
-  string(REPLACE "-" ";" GUID_VALS ${VIRTUALCAM_GUID})
-
-  list(LENGTH GUID_VALS GUID_VAL_COUNT)
-  if(GUID_VAL_COUNT EQUAL 5)
-    string(REPLACE ";" "0" GUID_HEX ${GUID_VALS})
-    string(REGEX MATCH "[0-9a-fA-F]+" GUID_ACTUAL_HEX ${GUID_HEX})
-    if(GUID_ACTUAL_HEX STREQUAL GUID_HEX)
-      list(GET GUID_VALS 0 GUID_VALS_DATA1)
-      list(GET GUID_VALS 1 GUID_VALS_DATA2)
-      list(GET GUID_VALS 2 GUID_VALS_DATA3)
-      list(GET GUID_VALS 3 GUID_VALS_DATA4)
-      list(GET GUID_VALS 4 GUID_VALS_DATA5)
-      string(LENGTH ${GUID_VALS_DATA1} GUID_VALS_DATA1_LENGTH)
-      string(LENGTH ${GUID_VALS_DATA2} GUID_VALS_DATA2_LENGTH)
-      string(LENGTH ${GUID_VALS_DATA3} GUID_VALS_DATA3_LENGTH)
-      string(LENGTH ${GUID_VALS_DATA4} GUID_VALS_DATA4_LENGTH)
-      string(LENGTH ${GUID_VALS_DATA5} GUID_VALS_DATA5_LENGTH)
-      if(GUID_VALS_DATA1_LENGTH EQUAL 8
-         AND GUID_VALS_DATA2_LENGTH EQUAL 4
-         AND GUID_VALS_DATA3_LENGTH EQUAL 4
-         AND GUID_VALS_DATA4_LENGTH EQUAL 4
-         AND GUID_VALS_DATA5_LENGTH EQUAL 12)
-        set(GUID_VAL01 ${GUID_VALS_DATA1})
-        set(GUID_VAL02 ${GUID_VALS_DATA2})
-        set(GUID_VAL03 ${GUID_VALS_DATA3})
-        string(SUBSTRING ${GUID_VALS_DATA4} 0 2 GUID_VAL04)
-        string(SUBSTRING ${GUID_VALS_DATA4} 2 2 GUID_VAL05)
-        string(SUBSTRING ${GUID_VALS_DATA5} 0 2 GUID_VAL06)
-        string(SUBSTRING ${GUID_VALS_DATA5} 2 2 GUID_VAL07)
-        string(SUBSTRING ${GUID_VALS_DATA5} 4 2 GUID_VAL08)
-        string(SUBSTRING ${GUID_VALS_DATA5} 6 2 GUID_VAL09)
-        string(SUBSTRING ${GUID_VALS_DATA5} 8 2 GUID_VAL10)
-        string(SUBSTRING ${GUID_VALS_DATA5} 10 2 GUID_VAL11)
-        set(VIRTUALCAM_AVAILABLE ON)
-        set(INVALID_GUID OFF)
+if(ENABLE_VIRTUALCAM)
+  if(VIRTUALCAM_GUID STREQUAL "")
+    obs_status(
+      WARNING
+      "Windows Virtual Camera - GUID not set - specify as 'VIRTUALCAM_GUID' to enable."
+    )
+  else()
```

Comment:
If the entire GUID check is guarded behind the `ENABLE` flag, then the `if(INVALID_GUID)` check on line 131 could be moved inside as well.

Doesn't make sense to warn about an invalid GUID if the feature is disabled in the first place.

Reply:
True, I missed that part, I'll make an additional commit with that added in.